docs/data/data-grid/rows/rows.md

```diff
@@ -279,28 +279,71 @@ For now, row reordering is disabled if sorting is applied to the grid.
 In addition, if row grouping or tree data is being used, the row reordering is also disabled.
 :::
 
-## 🚧 Row spanning
+## Row pinning [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+Pinned (or frozen, locked or floating) rows are rows that are visible at all times while the user scrolls the grid vertically.
+Pinned rows are not affected by sorting or filtering.
 
 :::warning
-This feature isn't implemented yet. It's coming.
+This feature is experimental, it needs to be explicitly activated using the `rowPinning` experimental feature flag.
+
+```tsx
+<DataGridPro experimentalFeatures={{ rowPinning: true }} {...otherProps} />
+```
 
-👍 Upvote [issue #207](https://github.com/mui/mui-x/issues/207) if you want to see it land faster.
 :::
 
-Each cell takes up the width of one row.
-Row spanning allows to change this default behavior.
-It allows cells to span multiple rows.
-This is very close to the "row spanning" in an HTML `<table>`.
+You can pin rows at the top or bottom of the grid by passing pinned rows data through the `pinnedRows` prop:
+
+```tsx
+const pinnedRows: GridPinnedRowsProp = {
+  top: [{ id: 0, brand: 'Nike' }],
+  bottom: [
+    { id: 1, brand: 'Adidas' },
+    { id: 2, brand: 'Puma' },
+  ],
+};
+
+<DataGridPro pinnedRows={pinnedRows} />;
+```
+
+:::warning
+Just like the `rows` prop, `pinnedRows` prop should keep the same reference between two renders.
+Otherwise, the grid will re-apply heavy work like sorting and filtering.
+:::
+
+The data format for pinned rows is the same as for the `rows` prop (see [Feeding data](/x/react-data-grid/rows/#feeding-data)).
+
+Pinned rows data should also meet [Row identifier](/x/react-data-grid/rows/#row-identifier) requirements.
+
+{{"demo": "RowPinning.js", "disableAd": true, "bg": "inline"}}
+
+### Controlling pinned rows
+
+You can control which rows are pinned by changing `pinnedRows`.
+
+In the demo below we use `actions` column type to add buttons to pin a row either at the top or bottom and change `pinnedRows` prop dynamically.
+
+{{"demo": "RowPinningWithActions.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+
+### Usage with pagination
```

Comment:
I think the pagination demo is interesting.

But I'm not sure we need a section for it specifically.
Is there something specific to document here besides that the pinned rows are not affected by pagination?

Reply:
The main goal was to demonstrate the behavior of pinned rows staying pinned on page / page size change.
But I can remove this section if you find it unnecessary.

Reply:
I think we can leave the demo but maybe rename the title too

```suggestion
### Usage with other features
```

And bring the :::info about the support to other features here.